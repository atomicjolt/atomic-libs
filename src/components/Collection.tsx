import {
  Item as ReactStatelyItem,
  Section as ReactStatelySection,
} from "react-stately";
import { ItemProps, SectionProps } from "react-stately";
import { cloneComponent } from "../clone";

// We re-define the react-stately Item and Section components
// Because for some reason storybook doesn't like to pickup the
// correct names for the reac-stately components.

/** An Item in a Collection component
 * @example
 * <CustomSelect label="Select a fruit">
 *  <Item key="apple">Apple</Item>
 *  <Item key="orange">Orange</Item>
 *  <Item key="banana">Banana</Item>
 * </CustomSelect>
 */
export const Item = cloneComponent(ReactStatelyItem, "Item");

/** A Section / grouping of items in a collection component
 * @example
 * <ListBox>
 *  <Section title="Fruits">
 *    <Item key="apple">Apple</Item>
 *    <Item key="orange">Orange</Item>
 *    <Item key="banana">Banana</Item>
 *  </Section>
 *  <Section title="Vegetables">
 *    <Item key="carrot">Carrot</Item>
 *    <Item key="potato">Potato</Item>
 *    <Item key="onion">Onion</Item>
 *  </Section>
 * </ListBox>
 */
export const Section = cloneComponent(ReactStatelySection, "Section");

export type { ItemProps, SectionProps };
