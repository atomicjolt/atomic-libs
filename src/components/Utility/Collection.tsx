import {
  Item as ReactStatelyItem,
  Section as ReactStatelySection,
  ItemProps,
  SectionProps,
} from "react-stately";

// We re-define the react-stately Item and Section components
// Because for some reason storybook doesn't like to pickup the
// correct names for the reac-stately components.

/** An Item in a Collection component
 * @example
 * <CustomSelect label="Select a fruit">
 *  <Item key="apple">Apple</Item>
 *  <Item key="orange">Orange</Item>
 *  <Item key="banana">Banana</Item>
 * </CustomSelect>
 */
export function Item<T>(props: ItemProps<T>) {
  return null;
}

// React Stately binds some static properties to the Item component that we need to copy over
for (const key in ReactStatelyItem) {
  // @ts-ignore
  Item[key] = ReactStatelyItem[key];
}

/** A Section / grouping of items in a collection component
 * @example
 * <ListBox>
 *  <Section title="Fruits">
 *    <Item key="apple">Apple</Item>
 *    <Item key="orange">Orange</Item>
 *    <Item key="banana">Banana</Item>
 *  </Section>
 *  <Section title="Vegetables">
 *    <Item key="carrot">Carrot</Item>
 *    <Item key="potato">Potato</Item>
 *    <Item key="onion">Onion</Item>
 *  </Section>
 * </ListBox>
 */
export function Section<T>(props: SectionProps<T>) {
  return null;
}

for (const key in ReactStatelySection) {
  // @ts-ignore
  Section[key] = ReactStatelySection[key];
}
