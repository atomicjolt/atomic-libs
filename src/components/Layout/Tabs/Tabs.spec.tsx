import React from "react";
import { test, expect } from "vitest";
import { render } from "@testing-library/react";
import { Tabs } from ".";
import { Item } from "../../Collection";

test("matches snapshot", () => {
  const res = render(
    <Tabs selectedKey="tab1">
      <Tabs.List>
        <Item id="tab1">
          <h1>Tab One</h1>
        </Item>
        <Item id="tab2">
          <h1>Tab Two</h1>
        </Item>
        <Item id="tab3">
          <h1>Tab Three</h1>
        </Item>
      </Tabs.List>
      <Tabs.Panel id="tab1">
        <h1>Tab One Content</h1>
      </Tabs.Panel>
      <Tabs.Panel id="tab2">
        <h1>Tab Two Content</h1>
      </Tabs.Panel>
      <Tabs.Panel id="tab3">
        <h1>Tab Three Content</h1>
      </Tabs.Panel>
    </Tabs>
  );

  expect(res.asFragment()).toMatchSnapshot();
});
