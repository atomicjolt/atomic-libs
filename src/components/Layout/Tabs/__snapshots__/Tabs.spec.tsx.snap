// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-kAycRU iOmgiD aje-tabs aje-tabs--default"
  >
    <div
      aria-orientation="horizontal"
      class="sc-aYaIB iOHgsp"
      id="react-aria-:r0:"
      role="tablist"
    >
      <div
        class="sc-gEvDqW kYrhgC"
      >
        <div
          class="sc-dcJtft cGwrWi"
        >
          <div
            aria-controls="react-aria-:r0:-tabpanel-tab1"
            aria-selected="true"
            class="sc-iGgVNO PPQuZ"
            data-key="tab1"
            id="react-aria-:r0:-tab-tab1"
            role="tab"
            tabindex="0"
          >
            Tab 1
          </div>
        </div>
        <div
          class="sc-dcJtft cGwrWi"
        >
          <div
            aria-selected="false"
            class="sc-iGgVNO PPQuZ"
            data-key="tab2"
            id="react-aria-:r0:-tab-tab2"
            role="tab"
            tabindex="-1"
          >
            Tab 2
          </div>
        </div>
        <div
          class="sc-dcJtft cGwrWi"
        >
          <div
            aria-selected="false"
            class="sc-iGgVNO PPQuZ"
            data-key="tab3"
            id="react-aria-:r0:-tab-tab3"
            role="tab"
            tabindex="-1"
          >
            Tab 3
          </div>
        </div>
      </div>
      <div
        class="sc-eqUzNf dAyzKR"
      />
    </div>
    <div
      aria-labelledby="react-aria-:r0:-tab-tab1"
      class="sc-gsFSjX hyLuVh"
      id="react-aria-:r0:-tabpanel-tab1"
      role="tabpanel"
      tabindex="0"
    >
      <h1>
        Tab One
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
